tools/clang/lib/SPIRV/SpirvContext.cpp

```diff
@@ -501,6 +510,31 @@ void SpirvContext::addDebugTypesToModule(SpirvModule *module) {
   for (const auto &typePair : typeTemplateParams) {
     module->addDebugInfo(typePair.second);
   }
+
+  debugTypes.clear();
+  typeTemplates.clear();
+  typeTemplateParams.clear();
+}
+
+template <>
+std::string
+SpirvContext::getDebugTypeHashValue<SpirvType>(const SpirvType *Val) {
+  return Val->getName();
```

Comment:
hashing types based on their names doesn't seem like a good idea. For example all structs will have the same name. I think you'll end up with chaining with a pretty long chain.

same for the following cases.

Am I missing something?